### PR TITLE
Docs: some improvements

### PR DESCRIPTION
**Pull request by valeriabarra**
This PR provides some doc improvements. Addresses #1034 

It fixes a figure linking problem. The space-filling curve was in the wrong path and was not found in the linking. Now it displays correctly (see screenshot of preview) 

image

And aims to provide solution of other warnings/issues.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by valeriabarra** [comment]
I think this is in a good shape now (pending my comment above). 

Merging this will close #1034

File: src/Topologies/Topologies.jl [hunk 1]
```diff
@@ -43,9 +44,9 @@ abstract type AbstractDistributedTopology <: AbstractTopology end
 coordinate_type(topology::AbstractTopology) = coordinate_type(domain(topology))
 
 """
-    domain(topology)
+    Topologies.domain(topology::AbstractTopology)
 
-Returns the domain of the `topology` from the underlying `mesh`
+Returns the [`Domains.AbstractDomain`](@ref) of the `topology` from the underlying `mesh`
```
Comment (charleskawczynski) [inline src/Topologies/Topologies.jl:49]:
```suggestion
Returns the domain (a subtype of [`Domains.AbstractDomain`](@ref)) of the `topology` from the underlying `mesh`
```

I might leave this as it was for preciseness, but we can include the ref and type information like this?

File: docs/src/index.md [hunk 2]
```diff
@@ -1 +1,11 @@
 # ClimaCore.jl
+
```
Comment (charleskawczynski) [inline docs/src/index.md:2]: Looks nice, thanks!

File: docs/src/installation_instructions.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Installation
+
```
Comment (charleskawczynski) [inline docs/src/installation_instructions.md:2]: This page is fine with me

File: docs/src/lib/ClimaCoreMakie.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# ClimaCoreMakie.jl
```
Comment (charleskawczynski) [inline docs/src/lib/ClimaCoreMakie.md:1]: LGTM

File: docs/src/lib/ClimaCorePlots.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# ClimaCorePlots.jl
```
Comment (charleskawczynski) [inline docs/src/lib/ClimaCorePlots.md:1]: LGTM

File: lib/ClimaCoreTempestRemap/src/onlineremap.jl [hunk 6]
```diff
@@ -2,7 +2,7 @@ import ClimaCore.DataLayouts: IJFH
 
 
 """
-    LinearTempestRemap{T, S, M, C, V}
+    ClimaCoreTempestRemap.LinearMap{T, S, M, C, V}
```
Comment (charleskawczynski) [inline lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: I don't think we need to qualify these methods in the doc string, Documenter already provides a qualified name above the doc string (see the API docs for an example).

The upside is that the name exactly matches, the downside is that all the function doc strings get a bit too wordy. Also, it's not a pattern that we're currently using across this and other repos (it doesn't seem to me to be a common pattern in other Julia packages either)

File: src/Topologies/Topologies.jl [hunk 7]
```diff
@@ -43,9 +44,9 @@ abstract type AbstractDistributedTopology <: AbstractTopology end
 coordinate_type(topology::AbstractTopology) = coordinate_type(domain(topology))
 
 """
-    domain(topology)
+    Topologies.domain(topology::AbstractTopology)
```
Comment (charleskawczynski) [inline src/Topologies/Topologies.jl:47]:
```suggestion
    domain(topology::AbstractTopology)
```

File: lib/ClimaCoreTempestRemap/src/onlineremap.jl (hunk 6 above)
Comment (valeriabarra) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: This was one of the "missing docrstring" errors that I tried to chase down and saw that this was a frequent issue among other Documenter users and found that qualifying would solve it. It did solve it in this instance

File: src/Topologies/Topologies.jl (hunk 7 above)
Comment (valeriabarra) [reply src/Topologies/Topologies.jl:47]: This was again my attempt at resolving the error (see above [comment](https://github.com/CliMA/ClimaCore.jl/pull/1035#issuecomment-1341645780)). I found some threads that discuss this (e.g., see [this](https://discourse.julialang.org/t/calling-docs-twice-for-the-same-function-in-documenter-jl/37791))

File: lib/ClimaCoreTempestRemap/src/onlineremap.jl (hunk 6 above)
Comment (charleskawczynski) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: We can do qualifying in the docs API

Comment (valeriabarra) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: Yeah, but then again, it was my attempt, before discovering that actually the docstring had an outdated name! Before, apparently it used to be called `LinearTempestRemap` and then it was later changed to `LinearMap` and people forgot to update the docstring

Comment (valeriabarra) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: So it's probably ok now to remove it altogether. I am going to try

Comment (charleskawczynski) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: We should probably update https://github.com/CliMA/ClimaCore.jl/blob/main/docs/make.jl#L59 to [:example_block, :export] (if that's the syntax), to ensure this doesn't keep popping up.

**Comment by simonbyrne** [comment]
For the topology image, can we plot the whole thing as one line?

**Comment by sriharshakandala** [comment]
I can modify it per our requirements. I just pushed temporary branch with the plotting code at https://github.com/CliMA/GilbertCurves.jl/tree/sk/cubed_sphere_sfc

File: lib/ClimaCoreTempestRemap/src/onlineremap.jl (hunk 6 above)
Comment (valeriabarra) [reply lib/ClimaCoreTempestRemap/src/onlineremap.jl:5]: @charleskawczynski , According to [Documenter's docs](https://documenter.juliadocs.org/stable/lib/public/index.html#Documenter.makedocs), the keywords are: `:autodocs_block`, `:cross_references`, `:docs_block`, `:doctest`, `:eval_block`, `:example_block`, `:footnote`, `:linkcheck`, `:meta_block`, `:missing_docs`, `:parse_error`, and `:setup_block`. So there is no `:export` or similar. What did you have in mind?

File: src/Topologies/Topologies.jl (hunk 7 above)
Comment (valeriabarra) [reply src/Topologies/Topologies.jl]: @charleskawczynski , I have applied the suggestion, but the problem is still present (see my comment in the PR convo)

**Comment by simonbyrne** [comment]
actually, let me add some numbering

**Comment by valeriabarra** [comment]
Thanks @simonbyrne . I updated the image.

**Comment by valeriabarra** [comment]
> actually, let me add some numbering

Ok, I'll update again

**Comment by simonbyrne** [comment]
```julia
using GilbertCurves, Plots

n = 8

panel_right = gilbertindices((n,n), majdim=1)
panel_up = gilbertindices((n,n), majdim=2)

list = vcat(panel_right,
            [idx + CartesianIndex(n,0) for idx in panel_up],
            [idx + CartesianIndex(n,n) for idx in panel_right],
            [idx + CartesianIndex(2n,n) for idx in panel_up],
            [idx + CartesianIndex(2n,2n) for idx in panel_right],
            [idx + CartesianIndex(3n,2n) for idx in panel_up])


p = plot([(c[1]-0.5)/n for c in list], [(c[2]-0.5)/n for c in list], line_z=1:length(list), legend=false, xaxis=false, yaxis=false, grid=false, alpha=0.5)

plot!(p,
      [1,2,2,3,3,4,4,3,3,2,2,1,1,0,0,1,1,2,2,3],
      [0,0,1,1,2,2,3,3,2,2,1,1,0,0,1,1,2,2,3,3];
      color=:black)

annotate!(p, 0.5,0.5, text("1", 30))
annotate!(p, 1.5,0.5, text("2", 30))
annotate!(p, 1.5,1.5, text("3", 30))
annotate!(p, 2.5,1.5, text("4", 30))
annotate!(p, 2.5,2.5, text("5", 30))
annotate!(p, 3.5,2.5, text("6", 30))

png(p, "panel.png")
```
panel

**Review by simonbyrne — APPROVED** [review]

File: src/Topologies/Topologies.jl (hunk 7 above)
Comment (charleskawczynski) [reply src/Topologies/Topologies.jl]: Ah, I see, we're importing the function and then re-defining the function stub and documenting that one. That actually seems a bit odd to me, @simonbyrne, should we just remove the duplicate function stub or leave it and document both qualified functions (e.g., `Topologies.domain(topology::AbstractTopology)`).

I suppose it's benign, so I'm fine with either one

**Review by charleskawczynski — APPROVED** [review]

**Comment by simonbyrne** [comment]
@valeriabarra feel free to merge whenever you're happy with it

File: src/Topologies/Topologies.jl (hunk 7 above)
Comment (valeriabarra) [reply src/Topologies/Topologies.jl]: I ended up removing the duplicated stub docstring. There was no way that by qualifying it, Documenter would be happy with it.
